query/src/catalogs/database.rs

```diff
@@ -12,7 +12,51 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-pub trait Database: Sync + Send {
+use std::sync::Arc;
+
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_meta_types::CreateTableReq;
+use common_meta_types::DropTableReply;
+use common_meta_types::DropTableReq;
+use common_meta_types::MetaId;
+use common_meta_types::TableIdent;
+use common_meta_types::TableInfo;
+use common_meta_types::TableMeta;
+use dyn_clone::DynClone;
+
+use crate::catalogs::Table;
+
+#[async_trait::async_trait]
+pub trait Database: DynClone + Sync + Send {
     /// Database name.
     fn name(&self) -> &str;
+
+    // Get one table by db and table name.
+    async fn get_table(&self, db_name: &str, table_name: &str) -> Result<Arc<dyn Table>>;
+
+    async fn list_tables(&self, db_name: &str) -> Result<Vec<Arc<dyn Table>>>;
+
+    async fn get_table_meta_by_id(&self, table_id: MetaId) -> Result<(TableIdent, Arc<TableMeta>)>;
```

Comment:
Is `table_id` still globally unique?

If it is, `get_table_meta_by_id` should be a method of `catalog`.

Reply:
The `get_table_meta_by_id` defined here, but i do not see where it used, will try move to catalog